tests/cpp_integration_tests/test_main_model_static.cpp

```diff
@@ -19,13 +20,14 @@ TEST_CASE("Test main model static") {
     std::array<Idx, 4> const link_indptr = {0, 2, 4, 6};
     std::array<Idx, 4> const source_indptr = {0, 1, 2, 3};
     // dependent dataset
-    ConstDataset update_data_dependent;
-    update_data_dependent["link"] = ConstDataPointer{link.data(), link_indptr.data(), batches};
-    update_data_dependent["source"] = ConstDataPointer{source.data(), source_indptr.data(), batches};
+    ConstDataset update_data_dependent{true, batches, "update", meta_data::meta_data_gen::meta_data};
+    update_data_dependent.add_buffer("link", -1, link_indptr.back(), link_indptr.data(), link.data());
+    update_data_dependent.add_buffer("source", -1, source_indptr.back(), source_indptr.data(), source.data());
+
     // independent dataset
-    ConstDataset update_data_independent;
-    update_data_independent["link"] = ConstDataPointer{link.data(), link_indptr.data(), batches - 1};
-    update_data_independent["source"] = ConstDataPointer{source.data(), source_indptr.data(), batches - 1};
+    ConstDataset update_data_independent{true, batches - 1, "update", meta_data::meta_data_gen::meta_data};
+    update_data_independent.add_buffer("link", -1, link_indptr.crbegin()[1], link_indptr.data(), source.data());
+    update_data_independent.add_buffer("source", -1, source_indptr.crbegin()[1], source_indptr.data(), source.data());
```

Comment:
the `indptr.crbegin()[1]` is difficult to understand for people not familiar with sparse index pointers. do you think we should create a separate function for this?

Reply:
This is the only place with this. Let's do not over generalize things until it is needed.